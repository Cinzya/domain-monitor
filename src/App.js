import React, { Component } from 'react';
import './App.css';
import Navigation from './components/navigation';
import List from './components/list';
import Intro from './components/intro';


class App extends Component{
  
  render() {
    return (
        <div className="App">
            <main>
                <section>
                    <Intro />
                    
                    <List />
                </section>
            </main>
            <Navigation />
        </div>

    );
  }
}

export default App;
