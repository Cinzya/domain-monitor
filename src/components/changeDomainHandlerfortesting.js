const changeDomainHandler = (event) => {
    let domainName = "";
    domainName = event;
}

module.exports = { changeDomainHandler};