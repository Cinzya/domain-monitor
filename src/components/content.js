import React, { Component } from 'react';
import logo from '../img/logo.svg';
import zahnrad from '../img/zahnrad.svg';
import muelleimer from '../img/muelleimer.svg';


class Content extends Component{
    constructor (props){
        super(props)
        this.state = {
            apiKey: "at_XtU8CpRcPmD7AX6RWswtOOK0voVgH",
            domainName: "",
            domains: []
        }
        this.renderTableData = this.renderTableData.bind(this);
        this.changeDomainHandler = this.changeDomainHandler.bind(this);
        this.onSubmitHandler = this.onSubmitHandler.bind(this);
    }

    changeDomainHandler(event) {
        // domainName wird der Wert des Eingabefelds zugewiesen
        this.setState({
            domainName: event.target.value
        });
    }

    addTime() {
        let today = new Date();
        let date = today.getFullYear()+'
        let time = today.getHours()+ ":
        let dateTime = date+' '+time;
        return dateTime
    };

    addData(toAdd){
        // ID ins State geschrieben
        toAdd.id = Math.random();
        // aktuelle Zeit wird zugewiesen
        toAdd.checked = this.addTime();
        // Daten werden ins State übertragen
        let domains = [toAdd];
        this.setState({
            domains: domains
        });
        console.log(this.state.domains)
    }

    onSubmitHandler(event) {
		event.preventDefault();
        // Ausgabe in der Konsole der aktuellen Werte
        console.log(this.state); //this.addDomain(this.state);

        // HTTP Anfrage an API
        fetch( "https://domain-availability-api.whoisxmlapi.com/api/v1?apiKey=" + this.state.apiKey + "&domainName=" + this.state.domainName, {
            method: 'GET',
        })
        .then(response => {
            return response.json();
        })
            // Ausgabe in der Konsole der API Response
        .then(responseData => {
            console.log(responseData);
            this.addData(responseData);
            // https://www.freecodecamp.org/forum/t/parsing-json-from-api-response/275420
            //this.setState({DomainInfo}),
            // const {DomainInfo: {domainAvailability: ''}} = this.state,
            // console.log(domainAvailability),
            //this.setState({ DomainInfo: JSON.parse(DomainInfo) }) {domainName: ''}
        });

        this.domainName = this.state;
        const newDomain = '';

        // Eingabefeld wird geleert
        this.setState({
            domainName: newDomain
        });
    }

    renderTableData(){
       return this.state.domains.map(domain => {
            return (
                <tr key={domain.id}>
                <td><img src={zahnrad} /></td>
                <td>{domain.DomainInfo.domainName}</td>
                <td><span>{domain.DomainInfo.domainAvailability}</span></td>
                <td>{domain.checked}</td>
                <td></td>
                <td><img src={muelleimer} /></td>
            </tr>

            )
        })
    }


    render() {
      return (
            <div className="column-right">
                <div className="logo">
                    <h1><span>Domain</span> <span>Monitor</span></h1>
                    <img src={logo}
                        alt="Logo"/>
                </div>
                {/* // InputField Component */}
                    <form className="domain-eingabe"
                      //hier muss noch iwie {this.additem} hin
                    onSubmit={(event)=> {this.onSubmitHandler(event)}} >
                    <input type="text" id="eingabefeld" placeholder="Geben Sie hier Ihre Wunschdomain ein"
                        value={this.state.domainName} onChange={this.changeDomainHandler}/>
                    <input type="submit" className="button" value="Hinzufügen"/>
                </form> {/* /* <List entries={this.state.domains}/>
        div Elemente um form und um alles, damit es klappt*/}

                    {/*  List Component  */}
                    <div className="domain-liste">
                        <div className="table-scrollable">
                            <table>
                                <tbody>
                                <tr>
                                    <th className="symbole">Einstellungen</th>
                                    <th>Domain</th>
                                    <th>Status</th>
                                    <th>zuletzt geprüft</th>
                                    <th>hinzugefügt</th>
                                    <th className="symbole">Löschen</th>
                                </tr>
                                    {this.renderTableData()}
                                </tbody>
                            </table>
                        </div>
                    </div>
            </div>
      );
    }
  }

  export default Content;