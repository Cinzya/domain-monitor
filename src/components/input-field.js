import React from 'react';

export const InputField = (props) => (
    <form
        className="domain-eingabe"
        onSubmit={(event)=> {props.Submit(event)}}
    >

        <input
            type="text" id="eingabefeld"
            placeholder="Geben Sie hier Ihre Wunschdomain ein"
            value={props.domainName}
            onChange={props.changeDomain}
        />

        <input
            type="submit"
            className="button"
            value="HinzufÃ¼gen"
        />
    </form>
);