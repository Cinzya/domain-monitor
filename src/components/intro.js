import React, { Component } from 'react';

class Intro extends Component {
    render() {
        return(
            <div className="column-left">
               
                <h2>Was ist ein <br></br> Domain Monitor?</h2>
                <p>Domain Monitor ist ein Dienst, mit dem Benutzer den Registrierungsstatus von Domains überwachen
                    können. Der Service benachrichtigt Sie (per E-Mail, wenn überwachte Domains kurz vor dem Ablauf
                    stehen und ihren Status ändern. Mit Domain Monitor können Sie Ihre Domains im Auge behalten ... oder
                    die anderer! Ablaufdaten, Statusänderungen - geben Sie einfach die Domainnamen ein, die Sie
                    überwachen möchten, und das Tool benachrichtigt Sie laufend über wichtige Änderungen bei der
                    Registrierung. Dieser Dienst kann sechs wichtige TLDs überwachen: .com, .org,, info, .us, .net und
                    .biz.</p>
                

            </div>
        );
    };
}

export default Intro;
