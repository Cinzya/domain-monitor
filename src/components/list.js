import React, { Component } from 'react';
import { connect } from 'react-redux';

import { Row } from './row';

import { deleteEvent } from '../actions/deleteDomainAction';

class List extends Component{
    constructor (props){
        super(props);
        this.deleteEvent = this.deleteEvent.bind(this);
        this.renderTableData = this.renderTableData.bind(this);
        this.compare = this.compare.bind(this);
    }

    //Zeile wird gelöscht
    deleteEvent(index){
        this.props.deleteEvent(index);
    }

    renderTableData(){
        return this.props.domains.map((domain, index) => {
            return (
                <Row
                    id={domain.id}
                    url={domain.domainName}
                    availability={domain.domainAvailability}
                    checked={domain.timeChecked}
                    added={domain.timeAdded}
                    delete={this.deleteEvent.bind(this, index)}/>
            )
        })
    }

    compare(key) {
        return this.props.innerSort(key, key);
    }

    render() {
        const { error, loading, domains } = this.props;

        if (error) {
            return <div className="error">Oops! Uns ist ein Fehler unterlaufen. Versuche es später noch ein Mal.</div>;
        }

        if (loading) {
            return <div className="loading">Loading...</div>;
        }

        return(
                <div className="domain-liste">
                    <div className="table-scrollable">
                        <table>
                            <tbody>
                            <tr>
                                <th className="symbole">Einstellungen</th>
                                <th> <button onClick={() =>
                                    {
                                        const sorted = domains.sort(this.compare("domainName"));
                                        this.setState({
                                            domains: sorted
                                        });
                                    }
                                }> Domain </button></th>
                                <th> <button onClick={() => {
                                    const sorted = domains.sort(this.compare("domainAvailability"));
                                    this.setState({
                                        domains: sorted
                                    });
                                }}> Status </button> </th>
                                <th> <button onClick={() => {
                                    const sorted = domains.sort(this.compare("timeChecked"));
                                    this.setState({
                                        domains: sorted
                                    });
                                }}>zuetzt geprüft</button> </th>
                                <th> <button onClick={() => {
                                    const sorted = domains.sort(this.compare("timeAdded"));
                                    this.setState({
                                        domains: sorted
                                    });
                                }}>hinzugefügt</button> </th>
                                <th className="symbole">Löschen</th>
                            </tr>
                            {this.renderTableData()}
                            </tbody>
                        </table>
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    domains: state.Domain.domains,
    loading: state.Domain.loading,
    error: state.Domain.error
});

const mapDispatchToProps = {
    deleteEvent
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(List);