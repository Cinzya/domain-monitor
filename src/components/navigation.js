import React, { Component } from 'react';

class Navigation extends Component {
    render() {
        return(
            <footer>
                <ul>
                    <li>Datenschutz</li>
                    <li>Impressum</li>
                </ul>
                <div className="clear"></div>
            </footer>
        );
    };
};

export default Navigation;