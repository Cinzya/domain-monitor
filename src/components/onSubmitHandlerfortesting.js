const onSubmitHandler = (event) => {
    let newDomain = '';
    newDomain = event;

}

module.exports = { onSubmitHandler };