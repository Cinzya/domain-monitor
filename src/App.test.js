import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount, render } from 'enzyme'; // Einbinden Enzyme
import { InputField } from './components/input-field';
import Intro from './components/intro'; // hinzugefügt
import List from './components/list'; // hinzugefügt
import Navigation from './components/navigation'; // hinzugefügt

//const wrapper = shallow(<Foo />);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

//Testing React Components
describe('InputField', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<InputField debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('Intro', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Intro debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('List', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<List debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('Navigation', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Navigation debug />);
    expect(component).toMatchSnapshot();
  });
});



