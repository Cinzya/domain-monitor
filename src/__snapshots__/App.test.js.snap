// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Intro should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`List should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Navigation should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
